File: client/src/app/shared/interfaces/furniture.ts
import {FurnitureType} from "./furniture-type";

export interface Furniture {
  title: string
  form: string
  price: number
  length: number
  width: number
  height: number
  diameter: number
  description: string
  furnitureType: FurnitureType
  priority: number
}

File: server/src/main/java/com/shikkeram/server/models/Furniture.java
package com.shikkeram.server.models;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Entity
@Table(name = "Furniture")
@Getter
@Setter
@NoArgsConstructor
public class Furniture extends AbstractEntity {
    private String title;
    @Column(length = 800)
    private String description;
    private Double length;
    private Double width;
    private Double height;
    private Double diameter;
    private String form;
    private Integer price;
    private Integer priority;
    @ManyToOne
    @JoinColumn(name = "furniture_type_id", referencedColumnName = "id")
    private FurnitureType furnitureType;
    @OneToMany(mappedBy = "furniture")
    private List<Image> images;
}
